frontend/app/mocks/subscription-api.server.ts

```diff
@@ -261,3 +330,53 @@ export function getSubscriptionApiMockHandlers() {
     }),
   ];
 }
+
+/**
+ * Retrieves a user entity based on the provided user ID.
+ *
+ * @param id - The user ID to look up in the database.
+ * @returns The user entity if found, otherwise throws a 404 error.
+ */
+function getUserEntity(id: string | readonly string[]) {
+  const parsedUserId = z.string().uuid().safeParse(id);
+  const userEntity = !parsedUserId.success
+    ? undefined
+    : db.user.findFirst({
+        where: { id: { equals: parsedUserId.data } },
+      });
+
+  if (!userEntity) {
+    throw new HttpResponse('User Not found', { status: 404, headers: { 'Content-Type': 'text/plain' } });
+  }
+
+  return userEntity;
+}
+
+/**
+ * Converts a user entity to a patch document with specific fields.
+ *
+ * @param userEntity - The user entity to convert.
+ * @returns Patch document containing selected user fields.
+ */
+function toUserPatchDocument({ email }: ReturnType<typeof getUserEntity>) {
+  return { email };
+}
+
+function getSubscriptionEntity(id: string | readonly string[]) {
+  const parsedSubscriptionId = z.string().safeParse(id);
+  const subscriptionEntity = !parsedSubscriptionId.success
+    ? undefined
+    : db.subscription.findFirst({
+        where: { id: { equals: parsedSubscriptionId.data } },
+      });
```

Comment:
A bit of a nitpick, but it's usually better to use positive code over negative code for readability:

``` typescript
  const subscriptionEntity = parsedSubscriptionId.success
    ? db.subscription.findFirst({
        where: { id: { equals: parsedSubscriptionId.data } },
      })
    : undefined;
```

Also, AI told me to use this.. not sure if the logic is the same though:

``` typescript
const subscriptionEntity = db.subscription.findFirst({
  where: { id: { equals: parsedSubscriptionId.data } },
}) ?? undefined;
```

Reply:
fixed